src/main/common/filter.h

```diff
@@ -54,20 +56,32 @@ typedef struct firFilterDenoise_s {
 } firFilterDenoise_t;
 
 typedef struct fastKalman_s {
-    float q;       // process noise covariance
-    float r;       // measurement noise covariance
-    float p;       // estimation error covariance matrix
-    float k;       // kalman gain
+    float k;       // "kalman" gain
     float x;       // state
     float lastX;   // previous state
 } fastKalman_t;
 
+typedef struct laggedMovingAverage_s {
+    int movingWindowIndex;
+    int windowSize;
+    float weight;
+    float movingSum;
+    float buf[MAX_LMA_WINDOW_SIZE];
+} laggedMovingAverage_t;
+
+// XXX(fujin): is there a way to combine these in some way? we could potentially
```

Comment:
Is should be quite easy ... 
It is confusing to split it, with stage 1, stage 2, d-term ...

Reply:
The main issue is with CLI, I don't think we want to provide the ability to use two BiquadRC+FIR2 or two FKF filters on stage1, stage2, and probably dterm.

Reply:
Why not ? ;-)

Bitmask may be used to limit allowed filters at given stage if necessary.

Reply:
Mainly to avoid more confusion, it will give 25 gyro lowpass filtering options for end-user to select from :)

Reply:
For testing best to have all the options.  End-user will stick with defaults mostly.